decode.go

```diff
@@ -870,10 +869,6 @@ func (d *decoder) mappingStruct(n *Node, out reflect.Value) (good bool) {
 		d.prepare(n, field)
 	}
 
-	var doneFields []bool
-	if d.uniqueKeys {
-		doneFields = make([]bool, len(sinfo.FieldsList))
```

Comment:
Why is this (and below) completely removed; I would have expected `terrors` to just swapped to the strict variant?